File: 송하연/10월 2주차/타겟넘버.java
import java.util.*;

class Solution {
    
    public static int cnt = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;

        int sum=0;
        int depth=0;
        int totalDepth = numbers.length-1;
        
        dfs(numbers, sum, target, depth, totalDepth); 

        answer = cnt;
        return answer;
    }

    private void dfs(int[] numbers, int sum, int target, int depth, int totalDepth){
        
        int plusSum = sum + numbers[depth];
        int minusSum = sum - numbers[depth];
        
        if(depth == totalDepth) {
            if(plusSum == target || minusSum == target)
            cnt++;
            return;
        }
        
        dfs(numbers, plusSum, target, depth+1, totalDepth);
        dfs(numbers, minusSum, target, depth+1, totalDepth);
       
    }
   
}

 
    

File: 한승우/10월 2주차/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int target;

int dfs(int cal, int idx, const vector<int>& numbers){
    if(idx == numbers.size()){
        if(cal == target) return 1; //하나 찾음
        return 0;
    }
    
    return dfs(cal + numbers[idx], idx + 1, numbers) + dfs(cal - numbers[idx], idx + 1, numbers);
}

int solution(vector<int> numbers, int tg) {
    int answer = 0; 
    target = tg;
    
    answer = dfs(0, 0, numbers);
    
    return answer;
}